### DISCUSSION TITLE
How to detect RPM version without running rpm --version?

### DISCUSSION THREAD

**Discussion by shanebishop** [rpm-software-management/rpm#2213] [Q&A] [answered]
How can the RPM version be detected programmatically (in a language like Go or C) using either a file or a system call, without running a subprocess `rpm --version`?

For context, I am writing an application that needs to be highly performant, and I want to avoid spawning any child processes if at all possible.

**Comment by pmatilai** [comment, marked as answer]
librpm does export its version:
> rpmlib.h:extern const char * const RPMVERSION;

But a highly performant application will remain that way if you call an external process, once.
